packages/visual-editor/src/components/pageSections/InsightSection.tsx

```diff
@@ -0,0 +1,233 @@
+import {
+  Image,
+  HeadingLevel,
+  BackgroundStyle,
+  YextField,
+  YextEntityField,
+  useDocument,
+  resolveYextEntityField,
+  PageSection,
+  Body,
+  Heading,
+  EntityField,
+  Background,
+  CTA,
+  backgroundColors,
+  VisibilityWrapper,
+} from "@yext/visual-editor";
+import { CTA as CTAType, ImageType } from "@yext/pages-components";
+import { ComponentConfig, Fields } from "@measured/puck";
+import { Timestamp, TimestampOption } from "../atoms/timestamp.tsx";
+
+/** TODO remove types when spruce is ready */
+type Insights = Array<InsightStruct>;
+
+type InsightStruct = {
+  image?: ImageType;
+  name?: string; // single line text
+  category?: string; // single line text
+  publishTime?: dateTime; // lexon's dateTime
+  description?: RTF2;
+  CTA?: CTAType;
+};
+
+type dateTime = {
+  start: string; // ISO 8601
+};
+
+type RTF2 = {
+  html?: string;
+  json?: Record<string, any>;
+};
+/** end of hardcoded types */
+
+export interface InsightSectionProps {
+  styles: {
+    backgroundColor?: BackgroundStyle;
+    cardBackgroundColor?: BackgroundStyle;
+  };
+  sectionHeading: {
```

Comment:
data section - looks like we don't have it for all of these list-type components